tests/test_app/TestApp/ApplicationWithDefaultRoutes.php

```diff
@@ -37,7 +38,12 @@ class ApplicationWithDefaultRoutes extends BaseApplication
      */
     public function bootstrap(): void
     {
-        // Do nothing.
+        // Only load plugins defined in Configure.
+        if (Configure::check('Plugins.autoload')) {
```

Comment:
I wonder if we can formalize this in a better way. Add a method like `TestCase::pluginsToLoad()` which simply holds the plugin names and then update the `ContainerStubTrait::createApp()` method to load those plugins using the app instance it creates.